### PR TITLE
Browserify Support

### LINKED ISSUE #114: Make ethereum.js browserify compatible
_ethereum.js_ doesn't support NPM users that use browserify to make their programs available in the browser. At present, if you attempt to use browserify on an app that requires `ethereum.js` than you will find that it includes a Node.js implementation of XHR, rather than the browser's native XHR.

_ethereum.js_ seems to use `process.env.NODE_ENV`, [envify](https://github.com/hughsk/envify)  & [unreachable-branch-transform](https://github.com/zertosh/unreachable-branch-transform) to support having different code paths in Node.js and the browser. This seems much clunkier than using the [features built-in to browserify](http://geojason.info/2014/writing-javascript-for-the-server-and-browser-with-browserify/), but it's worse than that since:

  1. If you don't set `process.env.NODE_ENV = 'debug'` you end up with a Node.js version of XHR.
  1. If you **do** set `process.env.NODE_ENV = 'debug'` than the program errors because [bignumber.js](https://github.com/MikeMcl/bignumber.js/) is purposely not loaded in that case (something to do with Meteor.js?).

IMO, a number of changes should be made:

  1. [envify](https://github.com/hughsk/envify)  & [unreachable-branch-transform](https://github.com/zertosh/unreachable-branch-transform) should be dropped.
  1. The `browser` field should be used to cause the native XHR package to be loaded when using browserify.
  1. [bignumber.js](https://github.com/MikeMcl/bignumber.js/) should always be used, in both the browser and Node.js.

If you are happy in principle with these changes, then I'd be happy to raise a pull-request for this work?

### PR DESCRIPTION
**Pull request by dchambers**
Speculative pull-request to fix #114, but where we continue using the existing [xmlhttprequest](https://github.com/driverdan/node-XMLHttpRequest) library. No documentation changes made yet as I don't understand the implications of this change for Bower and Meteor users.

### CODE REVIEW

File: lib/web3/httpprovider.js [hunk 1]
```diff
@@ -21,9 +21,7 @@
  * @date 2014
  */
 
-if (process.env.NODE_ENV !== 'build') {
-        var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest; // jshint ignore:line
-}
+var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
```
Comment (debris) [inline lib/web3/httpprovider.js]: please add
```javascript
 // jshint ignore:line
```

to remove jshint error on travis

**Comment by debris** [comment]
this pull request will also require alteration of gulpfile.js
```diff
diff --git a/gulpfile.js b/gulpfile.js
index 2a37736..50f0fcf 100644
--- a/gulpfile.js
+++ b/gulpfile.js
@@ -23,7 +23,7 @@ var browserifyOptions = {
     debug: true,
     insert_global_vars: false, // jshint ignore:line
     detectGlobals: false,
-    bundleExternal: false
+    bundleExternal: true
 };

 gulp.task('bower', function(cb){
@@ -47,6 +47,7 @@ gulp.task('build', ['clean'], function () {
     return browserify(browserifyOptions)
         .require('./' + src + '.js', {expose: 'web3'})
         .add('./' + src + '.js')
+        .ignore('crypto')
         .bundle()
         .pipe(exorcist(path.join( DEST, dst + '.js.map')))
         .pipe(source(dst + '.js'))
```

**Comment by debris** [comment]
Please also rebase on develop branch ;)

**Comment by kumavis** [comment]
@debris whats the ignore crypto for?

**Comment by debris** [comment]
`bignumber.js` uses node crypto in few additional methods. So we need 'ignore crypto' to prevent crypto from being browserified.

**Comment by dchambers** [comment]
Closing as this is now superseded by #120.

**Pull request closed without merging**